### PR TITLE
Personal Blog Templates: Vertical header, right aligned alternative

### PR DESCRIPTION
**Pull request by beafialho**
Adds Personal Blog Templates (Vertical header, right aligned) alternative listed in #132

Also exported as a zip file.

* [Preview loaded from this PR – available **before** this PR is merged](https://playground.wordpress.net/?import-site=https%3A%2F%2Fraw.githubusercontent.com%2FWordPress%2Ftwentytwentyfive%2Fplayground-changes-2024-08-28T14-12-52-194Z%2Ftwentytwentyfive%2Fplayground.zip)
* [Preview loaded from the main branch – available **after** this PR is merged](https://playground.wordpress.net/?import-site=https%3A%2F%2Fraw.githubusercontent.com%2FWordPress%2Ftwentytwentyfive%2Ftrunk%2Ftwentytwentyfive%2Fplayground.zip)

### CODE REVIEW

**Comment by beafialho** [comment]
Here is the code for the query loop pattern used in the blog, archive and search results templates:

```
<!-- wp:query {"queryId":4,"query":{"perPage":10,"pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":true,"taxQuery":null,"parents":[]}} -->
<div class="wp-block-query"><!-- wp:post-template -->
<!-- wp:group {"layout":{"type":"flex","orientation":"vertical","justifyContent":"stretch"}} -->
<div class="wp-block-group"><!-- wp:post-title {"isLink":true} /-->

<!-- wp:post-date {"fontSize":"x-small"} /--></div>
<!-- /wp:group -->

<!-- wp:spacer {"height":"var:preset|spacing|40"} -->
<div style="height:var(--wp--preset--spacing--40)" aria-hidden="true" class="wp-block-spacer"></div>
<!-- /wp:spacer -->

<!-- wp:post-excerpt {"moreText":""} /-->

<!-- wp:post-featured-image {"isLink":true,"aspectRatio":"16/9"} /-->

<!-- wp:spacer {"height":"var:preset|spacing|80"} -->
<div style="height:var(--wp--preset--spacing--80)" aria-hidden="true" class="wp-block-spacer"></div>
<!-- /wp:spacer -->
<!-- /wp:post-template -->

<!-- wp:query-pagination {"layout":{"type":"flex","justifyContent":"space-between","flexWrap":"wrap"}} -->
<!-- wp:query-pagination-previous /-->

<!-- wp:query-pagination-numbers /-->

<!-- wp:query-pagination-next /-->
<!-- /wp:query-pagination -->

<!-- wp:query-no-results -->
<!-- wp:paragraph {"placeholder":"Add text or blocks that will display when a query returns no results."} -->
<p>No posts were found.</p>
<!-- /wp:paragraph -->
<!-- /wp:query-no-results --></div>
<!-- /wp:query -->
```

**Comment by carolinan** [comment]
How can we make this design responsive?

**Comment by beafialho** [comment]
> How can we make this design responsive?

It's looking responsive on my end. The width of the query loop isn't ideal, but I tried to make the header width as small as possible so that you'd still be able to read the posts somewhat comfortably. Given the editor constraints, I don't see another way to build this one. Open to suggestions on a different solution though.

https://github.com/user-attachments/assets/8f4bd546-f026-4b09-9ed2-425c01f86e41

**Comment by carolinan** [comment]
I did change the browser width below 400px, perhaps that was overkill 😄

File: patterns/vertical-header-right-aligned-archive-template.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * @since Twenty Twenty-Five 1.0
+ */
+
+?>
+<!-- wp:columns {"isStackedOnMobile":false,"style":{"spacing":{"padding":{"right":"0","left":"0","top":"0","bottom":"0"},"blockGap":{"left":"0"}}}} -->
+<div class="wp-block-columns is-not-stacked-on-mobile" style="padding-top:0;padding-right:0;padding-bottom:0;padding-left:0">
+	<!-- wp:column {"width":"8rem"} -->
+	<div class="wp-block-column" style="flex-basis:8rem">
+		<!-- wp:group {"align":"wide","style":{"position":{"type":"sticky","top":"0px"},"spacing":{"padding":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|30","right":"var:preset|spacing|30"}}},"backgroundColor":"base","layout":{"type":"default"}} -->
+		<div class="wp-block-group alignwide has-base-background-color has-background" style="padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--30);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--30)">
```
Comment (juanfra) [inline patterns/vertical-header-right-aligned-archive-template.php]:
```suggestion
		<!-- wp:group {"align":"wide","style":{"position":{"type":"sticky","top":"0px"},"spacing":{"padding":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|30","right":"var:preset|spacing|30"}}},"layout":{"type":"default"}} -->
		<div class="wp-block-group alignwide" style="padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--30);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--30)">
```

File: patterns/vertical-header-right-aligned-home-template.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ * @since Twenty Twenty-Five 1.0
+ */
+
+?>
+<!-- wp:columns {"isStackedOnMobile":false,"style":{"spacing":{"padding":{"right":"0","left":"0","top":"0","bottom":"0"},"blockGap":{"left":"0"}}}} -->
+<div class="wp-block-columns is-not-stacked-on-mobile" style="padding-top:0;padding-right:0;padding-bottom:0;padding-left:0">
+	<!-- wp:column {"width":"8rem"} -->
+	<div class="wp-block-column" style="flex-basis:8rem">
+		<!-- wp:group {"align":"wide","style":{"position":{"type":"sticky","top":"0px"},"spacing":{"padding":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|30","right":"var:preset|spacing|30"}}},"backgroundColor":"base","layout":{"type":"default"}} -->
+		<div class="wp-block-group alignwide has-base-background-color has-background" style="padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--30);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--30)">
```
Comment (juanfra) [inline patterns/vertical-header-right-aligned-home-template.php]:
```suggestion
		<!-- wp:group {"align":"wide","style":{"position":{"type":"sticky","top":"0px"},"spacing":{"padding":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|30","right":"var:preset|spacing|30"}}},"layout":{"type":"default"}} -->
		<div class="wp-block-group alignwide" style="padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--30);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--30)">
```

File: patterns/vertical-header-right-aligned-page-template.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ * @since Twenty Twenty-Five 1.0
+ */
+
+?>
+<!-- wp:columns {"isStackedOnMobile":false,"style":{"spacing":{"padding":{"right":"0","left":"0","top":"0","bottom":"0"},"blockGap":{"left":"0"}}}} -->
+<div class="wp-block-columns is-not-stacked-on-mobile" style="padding-top:0;padding-right:0;padding-bottom:0;padding-left:0">
+	<!-- wp:column {"width":"8rem"} -->
+	<div class="wp-block-column" style="flex-basis:8rem">
+		<!-- wp:group {"metadata":{"name":"Header"},"align":"wide","style":{"position":{"type":"sticky","top":"0px"},"spacing":{"padding":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|30","right":"var:preset|spacing|30"}}},"backgroundColor":"base","layout":{"type":"default"}} -->
+		<div class="wp-block-group alignwide has-base-background-color has-background" style="padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--30);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--30)">
```
Comment (juanfra) [inline patterns/vertical-header-right-aligned-page-template.php]:
```suggestion
		<!-- wp:group {"metadata":{"name":"Header"},"align":"wide","style":{"position":{"type":"sticky","top":"0px"},"spacing":{"padding":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|30","right":"var:preset|spacing|30"}}},"layout":{"type":"default"}} -->
		<div class="wp-block-group alignwide" style="padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--30);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--30)">
```

File: patterns/vertical-header-right-aligned-post-template.php [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+ * @since Twenty Twenty-Five 1.0
+ */
+
+?>
+<!-- wp:columns {"isStackedOnMobile":false,"style":{"spacing":{"padding":{"right":"0","left":"0","top":"0","bottom":"0"},"blockGap":{"left":"0"}}}} -->
+<div class="wp-block-columns is-not-stacked-on-mobile" style="padding-top:0;padding-right:0;padding-bottom:0;padding-left:0">
+	<!-- wp:column {"width":"8rem"} -->
+	<div class="wp-block-column" style="flex-basis:8rem">
+		<!-- wp:group {"metadata":{"name":"Header"},"align":"wide","style":{"position":{"type":"sticky","top":"0px"},"spacing":{"padding":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|30","right":"var:preset|spacing|30"}}},"backgroundColor":"base","layout":{"type":"default"}} -->
+		<div class="wp-block-group alignwide has-base-background-color has-background" style="padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--30);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--30)">
```
Comment (juanfra) [inline patterns/vertical-header-right-aligned-post-template.php]:
```suggestion
		<!-- wp:group {"metadata":{"name":"Header"},"align":"wide","style":{"position":{"type":"sticky","top":"0px"},"spacing":{"padding":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|30","right":"var:preset|spacing|30"}}},"layout":{"type":"default"}} -->
		<div class="wp-block-group alignwide" style="padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--30);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--30)">
```

File: patterns/vertical-header-right-aligned-search-results-template.php [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+ * @since Twenty Twenty-Five 1.0
+ */
+
+?>
+<!-- wp:columns {"isStackedOnMobile":false,"style":{"spacing":{"padding":{"right":"0","left":"0","top":"0","bottom":"0"},"blockGap":{"left":"0"}}}} -->
+<div class="wp-block-columns is-not-stacked-on-mobile" style="padding-top:0;padding-right:0;padding-bottom:0;padding-left:0">
+	<!-- wp:column {"width":"8rem"} -->
+	<div class="wp-block-column" style="flex-basis:8rem">
+		<!-- wp:group {"metadata":{"name":"Header"},"align":"wide","style":{"position":{"type":"sticky","top":"0px"},"spacing":{"padding":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|30","right":"var:preset|spacing|30"}}},"backgroundColor":"base","layout":{"type":"default"}} -->
+		<div class="wp-block-group alignwide has-base-background-color has-background" style="padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--30);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--30)">
```
Comment (juanfra) [inline patterns/vertical-header-right-aligned-search-results-template.php]:
```suggestion
		<!-- wp:group {"metadata":{"name":"Header"},"align":"wide","style":{"position":{"type":"sticky","top":"0px"},"spacing":{"padding":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|30","right":"var:preset|spacing|30"}}},"layout":{"type":"default"}} -->
		<div class="wp-block-group alignwide" style="padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--30);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--30)">
```

File: patterns/vertical-header-right-aligned-sidebar.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+ * Inserter: no
+ *
+ * @package WordPress
+ * @subpackage Twenty_Twenty_Five
+ * @since Twenty Twenty-Five 1.0
+ */
+
+?>
+<!-- wp:heading {"style":{"typography":{"fontStyle":"normal","fontWeight":"600","textTransform":"uppercase","letterSpacing":"1.6px"}},"fontSize":"x-small"} -->
+<h2 class="wp-block-heading has-x-small-font-size" style="font-style:normal;font-weight:600;letter-spacing:1.6px;text-transform:uppercase"><?php esc_html_e( 'Other posts', 'twentytwentyfive' ); ?></h2>
```
Comment (juanfra) [inline patterns/vertical-header-right-aligned-sidebar.php]: We don't have the `x-small` preset any more and it was moved to `small`.

```suggestion
<!-- wp:heading {"style":{"typography":{"fontStyle":"normal","fontWeight":"600","textTransform":"uppercase","letterSpacing":"1.6px"}},"fontSize":"small"} -->
<h2 class="wp-block-heading has-small-font-size" style="font-style:normal;font-weight:600;letter-spacing:1.6px;text-transform:uppercase"><?php esc_html_e( 'Other posts', 'twentytwentyfive' ); ?></h2>
```

File: patterns/vertical-header-right-aligned-sidebar.php [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+<div style="height:var(--wp--preset--spacing--40)" aria-hidden="true" class="wp-block-spacer"></div>
+<!-- /wp:spacer -->
+
+<!-- wp:query {"queryId":12,"query":{"perPage":4,"pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":false,"taxQuery":null,"parents":[]}} -->
+<div class="wp-block-query">
+	<!-- wp:post-template -->
+		<!-- wp:group {"style":{"spacing":{"blockGap":"var:preset|spacing|20"}},"layout":{"type":"flex","orientation":"vertical"}} -->
+		<div class="wp-block-group">
+			<!-- wp:post-title {"isLink":true,"fontSize":"medium"} /-->
+			<!-- wp:post-date {"fontSize":"x-small"} /-->
```
Comment (juanfra) [inline patterns/vertical-header-right-aligned-sidebar.php]: We don't have the `x-small` preset any more and it was moved to `small`.

```suggestion
			<!-- wp:post-date {"fontSize":"small"} /-->
```

File: patterns/vertical-header-right-aligned-post-template.php [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+		<!-- /wp:group -->
+		<!-- wp:post-featured-image {"aspectRatio":"16/9"} /-->
+		<!-- wp:group {"style":{"spacing":{"padding":{"right":"var:preset|spacing|50"}}},"layout":{"type":"default"}} -->
+		<div class="wp-block-group" style="padding-right:var(--wp--preset--spacing--50)">
+			<!-- wp:group {"style":{"spacing":{"padding":{"top":"var:preset|spacing|20","bottom":"var:preset|spacing|20"}}},"layout":{"type":"flex","flexWrap":"nowrap","justifyContent":"space-between"}} -->
+			<div class="wp-block-group" style="padding-top:var(--wp--preset--spacing--20);padding-bottom:var(--wp--preset--spacing--20)">
+				<!-- wp:group {"style":{"spacing":{"blockGap":"var:preset|spacing|20"}},"layout":{"type":"flex","flexWrap":"nowrap"}} -->
+				<div class="wp-block-group">
+					<!-- wp:avatar {"size":30,"style":{"border":{"radius":"100px"}}} /-->
+					<!-- wp:post-author-name {"fontSize":"x-small"} /-->
```
Comment (juanfra) [inline patterns/vertical-header-right-aligned-post-template.php]: We don't have the `x-small` preset any more and it was moved to `small`.
```suggestion
					<!-- wp:post-author-name {"fontSize":"small"} /-->
```

File: patterns/vertical-header-right-aligned-posts.php [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+ */
+
+?>
+<!-- wp:query {"query":{"perPage":10,"pages":0,"offset":0,"postType":"post","order":"desc","orderBy":"date","author":"","search":"","exclude":[],"sticky":"","inherit":true,"taxQuery":null,"parents":[]}} -->
+<div class="wp-block-query">
+	<!-- wp:post-template -->
+		<!-- wp:group {"layout":{"type":"flex","orientation":"vertical","justifyContent":"stretch"}} -->
+		<div class="wp-block-group">
+			<!-- wp:post-title {"isLink":true} /-->
+			<!-- wp:post-date {"fontSize":"x-small"} /-->
```
Comment (juanfra) [inline patterns/vertical-header-right-aligned-posts.php]: We don't have the `x-small` preset any more and it was moved to `small`.

```suggestion
			<!-- wp:post-date {"fontSize":"small"} /-->
```

File: patterns/vertical-header-right-aligned-post-template.php [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+	<div class="wp-block-column" style="padding-top:var(--wp--preset--spacing--50);padding-right:0;padding-bottom:var(--wp--preset--spacing--50);padding-left:var(--wp--preset--spacing--50);flex-basis:90%">
+		<!-- wp:group {"style":{"spacing":{"padding":{"right":"var:preset|spacing|50","left":"0"}}},"layout":{"type":"default"}} -->
+		<div class="wp-block-group" style="padding-right:var(--wp--preset--spacing--50);padding-left:0">
+			<!-- wp:spacer {"height":"var:preset|spacing|50"} -->
+			<div style="height:var(--wp--preset--spacing--50)" aria-hidden="true" class="wp-block-spacer"></div>
+			<!-- /wp:spacer -->
+			<!-- wp:group {"layout":{"type":"flex","flexWrap":"wrap","justifyContent":"space-between","verticalAlignment":"top"}} -->
+			<div class="wp-block-group">
+				<!-- wp:post-title {"style":{"layout":{"selfStretch":"fixed","flexSize":"70vw"}},"fontSize":"xx-large"} /-->
+				<!-- wp:post-date {"textAlign":"right","style":{"elements":{"link":{"color":{"text":"var:preset|color|contrast"}}}},"textColor":"contrast","fontSize":"x-small"} /-->
```
Comment (juanfra) [inline patterns/vertical-header-right-aligned-post-template.php]: We don't have the `x-small` preset any more and it was moved to `small`.

```suggestion
				<!-- wp:post-date {"textAlign":"right","style":{"elements":{"link":{"color":{"text":"var:preset|color|contrast"}}}},"textColor":"contrast","fontSize":"small"} /-->
```

**Review by juanfra — COMMENTED** [review]
Great work, @beafialho! It's looking really nice ✨ 

Now that we have the section styles, I believe we could leverage those to style the whole "vertical header" by removing the base color as background.

I believe it'd be a better UX if the menu items are aligned to the left, as they'll be closer to where the opening of the menu is triggered. _It'd be so nice to be able to customize more things of the overlay menu, but that's editor territory._

https://github.com/user-attachments/assets/234dbf83-b5c7-4b2b-b643-74d1b8b7784f

Finally, I left some suggestions to some pieces that were using the `x-small` preset, as it's no longer part of the codebase and it was defaulting to some strange sizes, like the following:

Screenshot 2024-09-02 at 19 26 40

Note: I'm assuming this layout is meant to be full-width, and that I should threat the differences between this and figma like the other templates.

File: patterns/vertical-header-right-aligned-post-template.php (hunk 10 above)
Comment (carolinan) [reply patterns/vertical-header-right-aligned-post-template.php]: Since this is the default color, it can be removed, too.

Comment (carolinan) [reply patterns/vertical-header-right-aligned-post-template.php]: oh never mind, the default color for post date is primary, not contrast.

**Pull request merged**